File: sample_app/src/main/java/uk/co/alt236/btlescan/ui/details/recyclerview/model/AdRecordItem.java
package uk.co.alt236.btlescan.ui.details.recyclerview.model;

import uk.co.alt236.bluetoothlelib.device.adrecord.AdRecord;
import uk.co.alt236.bluetoothlelib.util.AdRecordUtils;
import uk.co.alt236.btlescan.ui.common.recyclerview.RecyclerViewItem;

public class AdRecordItem implements RecyclerViewItem {

    private final String mTitle;
    private final byte[] mData;
    private final String mDataAsString;

    public AdRecordItem(final String title,
                        final AdRecord record) {
        mTitle = title;
        mData = record.getData();
        mDataAsString = AdRecordUtils.getRecordDataAsString(record);
    }

    public String getTitle() {
        return mTitle;
    }

    public byte[] getData() {
        return mData;
    }

    public String getDataAsString() {
        return mDataAsString;
    }
}

File: sample_app/src/main/java/uk/co/alt236/btlescan/ui/details/recyclerview/model/AdRecordItem.kt
package uk.co.alt236.btlescan.ui.details.recyclerview.model

import dev.alt236.bluetoothlelib.device.adrecord.AdRecord
import dev.alt236.bluetoothlelib.util.AdRecordUtils
import dev.alt236.bluetoothlelib.util.ByteUtils
import uk.co.alt236.btlescan.app.ui.view.details.model.AdRecordItem
import uk.co.alt236.btlescan.app.ui.view.recyclerview.RecyclerViewItem
import uk.co.alt236.btlescan.ext.ByteArrayExt.toCharString

class AdRecordItem(
    override val title: String,
    record: AdRecord,
) : RecyclerViewItem,
    AdRecordItem {
    override val data: ByteArray = record.data ?: ByteArray(0)
    override val dataAsString: String = AdRecordUtils.getRecordDataAsString(record)
    override val dataAsChars: String = data.toCharString()
    override val dataAsHexArray: String = ByteUtils.byteArrayToHexString(record.data)
}
